R/predict.ergm.R

```diff
@@ -0,0 +1,75 @@
+#' ERGM-based conditional tie probabilities
+#'
+#' This will calculate **conditional** tie probabilities for all dyads in `net`
+#' based on the ERGM model `fit`.
+#'
+#' @param fit fitted ERGM model object
+#' @param net network object
+#'
+#' @return 
+#' N x N matrix of conditional tie probabilities, where N is the number of nodes in `net`.
+#' 
+#' @method predict ergm
+#'
+#' @export
+
+predict.ergm <- function(fit, net) {
+  # Argument checking
+  stopifnot(inherits(fit, "ergm"))
+  stopifnot(inherits(net, "network"))
+  if( is.directed(fit$network) != is.directed(net) )
+    stop(
+      "the model was fit to ",
+      if(is.directed(fit$network)) "directed" else "undirected",
+      " network while `net` is ",
+      if(is.directed(net)) "directed" else "undirected"
+    )
+  # Fix/enformulate curved ERGMs if necessary
+  if(is.curved(fit)) {
+    fixed <- fix.curved(fit)
+    coeff <- fixed$theta
+    form <- fixed$formula
+  } else {
+    coeff <- coef(fit)
+    form <- fit$formula
+  }
+  n <-  network.size(net)
+  # Substitute network object in formula's environment
+  e <- rlang::env(net = net)
+  rlang::f_env(form) <- e
+  rlang::f_lhs(form) <- quote(net)
+  # Calculate sufficient stats in `net`
+  z <- summary(form)
+  result <- matrix(0, n, n)
+  # Loop over all dyads and calculate expits on toggle
+  if(is.directed(net)) {
+    iseq <- 1:n
+  } else {
+    iseq <- seq(1, n-1)
+  }
+  for(i in iseq) {
+    if(is.directed(net)) {
+      jseq <- setdiff(1:n, i)
+    } else {
+      jseq <- seq(i + 1, n)
+    }
+    for(j in jseq) {
```

Comment:
This is not a very efficient way to do this. `ergmMPLE(output="array")` can output the conditional predictor matrices for all the statistics.

Reply:
Great! I was looking for something like that.